pymc/tests/distributions/test_timeseries.py

```diff
@@ -481,56 +481,120 @@ def test_change_dist_size(self):
+        garch_like = t.compile_logp(y)({"y": data})
+        reg_like = t.compile_logp(z)({"z": data})
+        decimal = select_by_precision(float64=7, float32=4)
+        np.testing.assert_allclose(garch_like, reg_like, 10 ** (-decimal))
+
+    @pytest.mark.parametrize(
+        "arg_name",
+        ["omega", "alpha_1", "beta_1", "initial_vol"],
+    )
+    def test_batched_size(self, arg_name):
+        steps, batch_size = 100, 5
+        param_val = np.square(np.random.randn(batch_size))
+        init_kwargs = dict(
+            omega=1.25,
+            alpha_1=0.5,
+            beta_1=0.45,
+            initial_vol=2.5,
+        )
+        kwargs0 = init_kwargs.copy()
+        kwargs0[arg_name] = init_kwargs[arg_name] * param_val
+        with Model() as t0:
+            y = GARCH11("y", shape=(batch_size, steps), **kwargs0)
+
+        y_eval = draw(y, draws=2)
+        assert y_eval[0].shape == (batch_size, steps)
+        assert not np.any(np.isclose(y_eval[0], y_eval[1]))
+
+        kwargs1 = init_kwargs.copy()
+        with Model() as t1:
+            for i in range(batch_size):
+                kwargs1[arg_name] = init_kwargs[arg_name] * param_val[i]
+                GARCH11(f"y_{i}", shape=steps, **kwargs1)
+
+        np.testing.assert_allclose(
+            t0.compile_logp()(t0.initial_point()),
+            t1.compile_logp()(t1.initial_point()),
+        )
+
+    @pytest.mark.parametrize(
+        "size, expected",
         [
-            -1.35078362,
-            -0.81254164,
-            0.28918551,
-            -2.87043544,
-            -0.94353337,
-            0.83660719,
-            -0.23336562,
-            -0.58586298,
-            -1.36856736,
-            -1.60832975,
-            -1.31403141,
-            0.05446936,
-            -0.97213128,
-            -0.18928725,
-            1.62011258,
-            -0.95978616,
-            -2.06536047,
-            0.6556103,
-            -0.27816645,
-            -1.26413397,
-        ]
+            (None, np.zeros((2, 8))),
+            ((5, 2), np.zeros((5, 2, 8))),
+        ],
     )
-    omega = 0.6
-    alpha_1 = 0.4
-    beta_1 = 0.5
-    initial_vol = np.float64(0.9)
-    vol = np.empty_like(data)
-    vol[0] = initial_vol
-    for i in range(len(data) - 1):
-        vol[i + 1] = np.sqrt(omega + beta_1 * vol[i] ** 2 + alpha_1 * data[i] ** 2)
-
-    with Model() as t:
-        y = GARCH11(
-            "y",
-            omega=omega,
-            alpha_1=alpha_1,
-            beta_1=beta_1,
-            initial_vol=initial_vol,
-            shape=data.shape,
+    def test_moment(self, size, expected):
+        with Model() as model:
+            GARCH11(
+                "x",
+                omega=1.25,
+                alpha_1=0.5,
+                beta_1=0.45,
+                initial_vol=np.ones(2),
+                steps=7,
+                size=size,
+            )
+        assert_moment_is_expected(model, expected, check_finite_logp=False)
```

Comment:
Any reason for `check_finite_logp`. We usually only disable it when the distribution logp would fail (e.g., 3D MvNormal not implemented)

Reply:
I was copying the pattern in AR - enabling it here.

Reply:
That guy who refactored the AR was a dummy :D